airbyte-integrations/bases/base-singer/base_singer/singer_helpers.py

```diff
@@ -73,7 +90,22 @@ def singer_catalog_to_airbyte_catalog(singer_catalog: Dict[str, any]) -> Airbyte
         for stream in singer_catalog.get("streams"):
             name = stream.get("stream")
             schema = stream.get("schema")
-            airbyte_streams += [AirbyteStream(name=name, json_schema=schema)]
+            airbyte_stream = AirbyteStream(name=name, json_schema=schema)
+
+            metadatas = stream.get("metadata")
+            stream_metadata_container = get_stream_level_metadata(metadatas) if metadatas else None
+            stream_metadata = stream_metadata_container.get("metadata")
+            if stream_metadata:
+                # TODO unclear from the singer spec what behavior should be if there are no valid replication keys, but forced-replication-method is INCREMENTAL.
```

Comment:
i think this is the a good choice. i think most taps throw exceptions in this case.